### DISCUSSION TITLE
Question - how to deploy a VirtualBox VM in another host?

### DISCUSSION THREAD

**Discussion by zulhfreelancer** [terra-farm/terraform-provider-virtualbox#159] [General]
Consider you have these hosts in your network:

![image](https://user-images.githubusercontent.com/3416976/205259396-cd149f5b-3965-4fd5-9ccb-15076db0f529.png)

Is it possible to create a VirtualBox VM in 192.168.0.101 host from 192.168.0.100 host?

Do we have something that we can install (agent?) on 192.168.0.101 to listen for `terraform plan` / `terraform apply` commands from outside and create the requested VMs on the host?

Does https://github.com/terra-farm/go-virtualbox support this?

**Comment by Kraktorist** [comment]
go-virtualbox  is a wrapper package. It uses `VBoxManage` utility to interact with VirtualBox. As I know it's not possible to use VboxManage to control VirtualBox instances installed on remote hosts.

But you can do something like this
ssh user@192.168.0.101 terraform plan
ssh user@192.168.0.101 terraform apply

or maybe create a bash alias for this command.

**Reply by VoyTechnology** [reply]
To expand on this, it's probably even worse of a hack, but you can alias the `VBoxManage` itself to ssh into the remote box and run `VBoxManage` there.

```sh
alias VBoxManage='ssh 192.168.0.101 VBoxManage'
```

But it also might mean that you would need to copy the image you want to run, probably in the same directory. I'm just guessing here.

**Reply by zulhfreelancer** [reply]
Thanks both
